docs/modules/ROOT/pages/api/upgrades.adoc

```diff
@@ -75,6 +114,19 @@ Requirements:
 
 Emits an {Upgraded} event.
 
+[.contract-item]
+[[UpgradeableComponent-upgrade_and_call]]
+==== `[.contract-item-name]#++upgrade_and_call++#++(ref self: ContractState, new_class_hash: ClassHash, selector: felt252, calldata: Span<felt252>)++` [.item-kind]#internal#
+
+Replaces the contract's class hash with `new_class_hash` and then calls `selector`
+from the upgraded context.
+
```

Comment:
I would add a note mentioning that should be kept in mind that the call is executed from the upgraded contract itself, and not the calling account, since they may have some implications in the setup.

Also, another note that could be helpful is that with native account abstraction, and multicall being part of SNIP6, this behavior could be achieved by using just upgraded and putting the extra call in the second element of the list of calls. Then the call would be executed from the account itself, and can't be front-runned.

Reply:
Done!